server_management/management/commands/pulldb.py

```diff
@@ -1,53 +1,59 @@
-            # Dump the database on the server.
-            sudo("su - {user} -c 'pg_dump {name} -cOx -U {user} -f /home/{user}/{name}.sql --clean'".format(
-                name=remote['database']['name'],
-                user=remote['database']['user'],
-            ))
-
-            # Pull the SQL file down.
-            local('scp {} {}@{}:/home/{}/{}.sql ~/{}.sql'.format(
-                '' if not getattr(env, 'key_filename') else ' -i {} '.format(env.key_filename),
-                env.user,
-                env.host_string,
-                remote['database']['user'],
-                remote['database']['name'],
-                config['local']['database']['name'],
-            ))
-
-            # Delete the file on the server.
-            sudo('rm -f /home/{}/{}.sql'.format(
-                remote['database']['user'],
-                remote['database']['name'],
-            ))
-
-            # Drop the local db
+        config, connection = load_config(options.get('remote', ''), debug=options.get('debug', False))
+        remote = config['remotes'][config['remote_name']]
+
+
+        # Dump the database on the server.
+        connection.sudo("su - {user} -c 'pg_dump {name} -cOx -U {user} -f /home/{user}/{name}.sql --clean'".format(
+            name=remote['database']['name'],
+            user=remote['database']['user'],
+        ))
+
+        # Pull the SQL file down.
+        connection.local('scp{} {}@{}:/home/{}/{}.sql ~/{}.sql'.format(
+            f' -i {connection.connect_kwargs["key_filename"]}' if connection.connect_kwargs.get('key_filename', False) else '',
+            connection.user,
+            connection.host,
+            remote['database']['user'],
+            remote['database']['name'],
+            config['local']['database']['name'],
+        ))
+
+        # Delete the file on the server.
+        connection.sudo('rm -f /home/{}/{}.sql'.format(
+            remote['database']['user'],
+            remote['database']['name'],
+        ))
+
+        # Drop the local db
+        try:
+            # We use invoke's run() here (which we've imported as local() for readability) to maintain the PATH envvar so we have access to PSQL commands
             local('dropdb {}'.format(
                 config['local']['database']['name']
             ))
-
-            # Create a new db, this is an easy way to start fresh
-            local('createdb {}'.format(
-                config['local']['database']['name']
-            ))
-
-            # Import the database locally
-            local('psql -q {name} < ~/{name}.sql > /dev/null 2>&1'.format(
-                name=config['local']['database']['name'],
-            ))
-
-            # Cleanup local files
-            local('rm ~/{}.sql'.format(
-                config['local']['database']['name'],
-            ))
+        except UnexpectedExit as e:
+            if not f'''database "{config['local']['database']['name']}" does not exist''' in e.result.stderr:
```

Comment:
Better than making assumptions about how an error message will be formatted, would be to check the value of `e.result.exited` - if the exit code is 1 then it means the command did execute, but failed, which will probably mean you're trying to drop a database that does not exist (i.e. you can ignore it).

Exit code 1 is also used for dropping a database that is in use, but there's no exception handling around `createdb` below so it'll fail there instead.

Reply:
I was wanting to check for explicitly db not existing. I'm not certain if exit code 1 is only throw for this situation. What would happen if the user didn't have permissions to drop the chosen db? Would it be a different code?

Reply:
Nope, that would be exit code 1 as well, but that means that the `createdb` would fail noisily instead.